### PR TITLE
Cancel pending commit before starting on root

### PR DESCRIPTION
**Pull request by acdlite**
Moves the cancelTimeout call to right before creating a new work-in-progress root. Fixes a class of bugs where a pending commit is not cancelled, causing an incomplete tree to accidentally commit.

In the interest of fixing downstream bugs quickly, I'm landing this without a test case; I'll add one in a follow up.

### CODE REVIEW

File: packages/react-reconciler/src/__tests__/ReactSuspenseWithNoopRenderer-test.internal.js [hunk 1]
```diff
@@ -1724,11 +1724,3 @@ describe('ReactSuspenseWithNoopRenderer', () => {
     expect(ReactNoop.getChildren()).toEqual([span('Loading...')]);
   });
 });
-
-// TODO:
```
Comment (acdlite) [inline packages/react-reconciler/src/__tests__/ReactSuspenseWithNoopRenderer-test.internal.js]: This is unrelated. I happened to notice them now. Deleting these todos because they are outdated.

**Comment by acdlite** [comment]
Got a verbal approval from @bvaughn to land this as he was packing up to leave

**Comment by blling** [comment]
What does ’“leave” mean ?

**Comment by bvaughn** [comment]
Sometimes after working 8 or 9 hours, I go home to rest.
